internal/services/signalr/signalr_service_resource.go

```diff
@@ -341,122 +341,105 @@ func resourceArmSignalRServiceUpdate(d *pluginsdk.ResourceData, meta interface{}
 			}
+			resourceType.Properties.PublicNetworkAccess = utils.String(publicNetworkAcc)
+		}
 
-			if d.HasChange("identity") {
-				identity, err := identity.ExpandSystemOrUserAssignedMap(d.Get("identity").([]interface{}))
-				if err != nil {
-					return fmt.Errorf("expanding `identity`: %+v", err)
-				}
-				resourceType.Identity = identity
-			}
+		if d.HasChange("local_auth_enabled") {
+			resourceType.Properties.DisableLocalAuth = utils.Bool(!d.Get("local_auth_enabled").(bool))
+		}
 
-			if d.HasChanges("connectivity_logs_enabled", "messaging_logs_enabled", "service_mode", "live_trace_enabled", "http_request_logs_enabled") {
-				features := make([]signalr.SignalRFeature, 0)
-				logChanged := false
-				var connectivityLogsEnabled, messagingLogsEnabled, httpLogsEnabled bool
-				if d.HasChange("connectivity_logs_enabled") {
-					if v, ok := d.GetOk("connectivity_logs_enabled"); ok {
-						connectivityLogsEnabled = v.(bool)
-					}
-					features = append(features, signalRFeature(signalr.FeatureFlagsEnableConnectivityLogs, strconv.FormatBool(connectivityLogsEnabled)))
-					logChanged = true
-				}
+		if d.HasChange("aad_auth_enabled") {
+			resourceType.Properties.DisableAadAuth = utils.Bool(!d.Get("aad_auth_enabled").(bool))
+		}
 
-				if d.HasChange("messaging_logs_enabled") {
-					if v, ok := d.GetOk("messaging_logs_enabled"); ok {
-						messagingLogsEnabled = v.(bool)
-					}
-					features = append(features, signalRFeature(signalr.FeatureFlagsEnableMessagingLogs, strconv.FormatBool(messagingLogsEnabled)))
-					logChanged = true
-				}
+		if d.HasChange("tls_client_cert_enabled") {
+			tlsClientCertEnabled := d.Get("tls_client_cert_enabled").(bool)
+			resourceType.Properties.Tls = &signalr.SignalRTlsSettings{
+				ClientCertEnabled: utils.Bool(tlsClientCertEnabled),
+			}
+			if currentSku == "Free_F1" && tlsClientCertEnabled {
+				return fmt.Errorf("SKU Free_F1 does not support enabling tls client cert")
+			}
+		}
 
-				if d.HasChange("http_request_logs_enabled") {
-					if v, ok := d.GetOk("http_request_logs_enabled"); ok {
-						httpLogsEnabled = v.(bool)
-					}
-					logChanged = true
-				}
+		if d.HasChanges("connectivity_logs_enabled", "messaging_logs_enabled", "http_request_logs_enabled", "live_trace_enabled", "service_mode") {
+			features := make([]signalr.SignalRFeature, 0)
+			if d.HasChange("connectivity_logs_enabled") || d.HasChange("messaging_logs_enabled") || d.HasChange("http_request_logs_enabled") {
+				_, connectivityLogsNew := d.GetChange("connectivity_logs_enabled")
+				features = append(features, signalRFeature(signalr.FeatureFlagsEnableConnectivityLogs, strconv.FormatBool(connectivityLogsNew.(bool))))
+				_, messagingLogsNew := d.GetChange("messaging_logs_enabled")
+				features = append(features, signalRFeature(signalr.FeatureFlagsEnableMessagingLogs, strconv.FormatBool(messagingLogsNew.(bool))))
+				_, httpLogsNew := d.GetChange("http_request_logs_enabled")
```

Comment:
Thanks @stephybun , the three properties all belong to the resource log configuration. We need to call the expandResource Log function and pass the new values to the function if any of them changed. This is similar to the `live_trace`, the only difference is the three properties are not nested under `log_configuration` and we need to detect the change by using `|` logic.

The acctest is failed because we didn't include the properties into the haschanges list
![image](https://user-images.githubusercontent.com/92154856/229842668-9a3b6b41-e92c-4207-920f-4a19769b10d1.png)